keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -250,9 +406,30 @@ def __init__(
 
         super().__init__(**kwargs)
 
+        if vocabulary is None and lang is None:
+            raise ValueError(
+                "Tokenizer requires either the `vocabulary` or `lang` "
+                "argument. Use `vocabulary` for custom vocabulary and `lang` "
+                "for pre-trained vocabulary."
+            )
+        elif vocabulary is not None and lang is not None:
+            raise ValueError(
+                "Tokenizer requires only one of `vocabulary` or `lang` "
+                "arguments. Use `vocabulary` for custom vocabulary and `lang` "
+                "for pre-trained vocabulary."
+            )
+        elif vocabulary is None and lang is not None:
+            vocabulary = download_vocabulary(
```

Comment:
minor point, but can't we just return the dowloaded file path here? Then we don't have to duplicate the file loading line `line[:-1] for line in tf.io.gfile.GFile(vocabulary)`. Which will be nice if we need to update it.

Reply:
That was what I had previously too, but I had to change it for mock testing of the download() function. I had to have it return vocabularies instead of a filename to be able to create a useful mock.

Reply:
Got it! Sounds good!